content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,314 @@
+Maintaining a DAG improves efficiency, because a single vote on a DAG vertex implicitly votes for all transactions on the path to the genesis vertex.
+It also improves security, because the DAG interwines the fate of transactions, similar to the Bitcoin blockchain - this renders past decisions difficult to undo without the approval of correct nodes.
+
+When a client creates a transaction, it names one or more parent transactions, which are included inseparably in the inputs and their connections form the edges of the DAG (this parent-child relationship encoded in the DAG may or may not correspond to application specific dependencies - for example a child transaction need not spend ot have relationship with the funds received in parent transaction.
+All transactions reachable via parent edges back in history are called - `ancestor set` and all children transactions and their offspring are called `progeny`.
+
+The central challenge in the maintainance of the DAG is to choose among conflicting transactions.
+The notion of conflict is application-defined and transitive and forms an equivalence relation.
+For example transactions that spend the same funds form a conflict set and only one of them can be accepted.
+
+Avalanche instantiates a [Snowball](https://medium.com/@zaver.max/exploring-liveness-of-avalanche-d22f13b2db00) instance for each conflict set.
+Snowball uses repeated queries and counters to capture the amount of confidence built in conflicting transactions.
+
+In a given round each validator randomly selects a predefined number `n` of nodes from its validator list to query for their preferred decision on a value.
+Each queried validator responds with their preferred decisions, and if the majority of responses returned in a round differ from the one the node performing the query has, then it will update its own preferred decision to reflect that and respond to other nodes with that answer.
+All other validators process the same steps with subsamples of their validator lists.
+
+When a transaction is queried, all transactions reachable from it by following the DAG edges towards its ancestor set and its progeny are implicitly part of the query.
+A node will respond positively to the query if the transaction and its entire ancestry are currently the preferred option in their respective conflict sets.
+If more than a treshold of responders vote positively, the transaction is said to collect a `chit`.
+Nodes compute confidence in transactions as the total amount of chits they obtained - they query a transaction only once and rely on new vertices and possible chits added to the progeny to build up their confidence.
+
+TODO: add a picture of avalanche protocol steps
+
+Each of the nodes perform the querying round independently and asynchronously.
```

Comment:
```suggestion
Nodes perform the querying round independently and asynchronously.
```

Reply:
But why?

Reply:
Corrected here https://github.com/optakt/optakt.github.io/pull/21/commits/f524c8d94d32f6edd6da60a9e7f74a848c44d4ce#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R264, but is it clear that every single node performs the process on its own; I am not a native speaker and I am not sure, WDYT?

Reply:
When a sentence uses [general truth](https://www.grammaring.com/tags/grammar-tags/general-truth#:~:text=Present%20simple%20for%20general%20truths&text=The%20present%20simple%20is%20used,sun%20rises%20in%20the%20east.) like this, there's no need to specify that it's `Each of the nodes`. Saying that as a general truth, nodes do X, means that all nodes do X. I am not a native speaker either but I am 100% sure on this one 😄 

It's like saying: _magnets attract iron_. This sentence implicitly says that _all_ magnets attract iron. If something does not attract iron, it can not be qualified as a magnet.

Reply:
Cool, got it :) 
Thanks!